doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -914,15 +914,17 @@ async def generate_specific_payload_imagestreams(self, arch: str, private_mode:
         for payload_tag_name, payload_entry in payload_entries.items():
             multi_specs[private_mode].setdefault(payload_tag_name, dict())
 
-            if private_mode is False and payload_entry.build_inspector \
-                    and payload_entry.build_inspector.is_under_embargo():
-                # No embargoed images should go to the public release controller, so we will not have
+            if (private_mode is False and payload_entry.build_inspector
+               and payload_entry.build_inspector.is_under_embargo()
+               and self.runtime.assembly_type == AssemblyTypes.STREAM):
+                # No embargoed images for assembly stream
+                # should go to the public release controller, so we will not have
                 # a complete set of payload tags for the public imagestream.
                 incomplete_payload_update = True
+                continue
```

Comment:
Not sure if giving up an `else` clause in favor of a `continue` makes the code clearer. IIUC the behavior is the same (only `and self.runtime.assembly_type == AssemblyTypes.STREAM)` was added)